modules/userId/index.js

```diff
@@ -616,6 +631,95 @@ function getUserIdsAsEids() {
   return createEidsArray(getCombinedSubmoduleIds(initializedSubmodules));
 }
 
+/**
+ * This function will be exposed in global-name-space so that userIds stored by Prebid UserId module can be used by external codes as well.
+ * Simple use case will be passing these UserIds to A9 wrapper solution
+ */
+
+function getUserIdsAsEidBySource(sourceName) {
+  initializeSubmodulesAndExecuteCallbacks();
+  return createEidsArray(getSubmoduleId(initializedSubmodules, sourceName));
+};
+
+/**
+ * This function will be exposed in global-name-space so that userIds for a source can be exposed
+ * Sample use case is exposing this function to ESP
+ */
+function getEncryptedEidsForSource(source, encrypt, customFunction) {
+  let eids = [];
+  let eidsSignals = {};
+
+  if (isFn(customFunction)) {
+    logInfo(`${MODULE_NAME} - Getting encrypted signal from custom function : ${customFunction.name} & source : ${source} `);
+    // Publishers are expected to define a common function which will be proxy for signal function.
+    let customSignals = customFunction(source);
+    eidsSignals[source] = customSignals ? encryptSignals(customSignals) : null; // by default encrypt using base64 to avoid JSON errors
+  } else {
+    // initialize signal with eids by default
+    eids = getUserIdsAsEidBySource(source);
+    logInfo(`${MODULE_NAME} - Getting encrypted signal for eids :${JSON.stringify(eids)}`);
+    eids.forEach((eid) => {
```

Comment:
Earlier it use to return array of object with eids of particular source. Now it will return the object of eid for that particular source. Removed forEach